### PR TITLE
Only Run Local Fork For Dev Scenarios

### LINKED ISSUE #162: Only Run Local Fork For Dev Scenarios
**Background:**
Currently, spinning up a fork of the network is done with a special command: `poetry run start-fork`.
This is great for simulating executing transactions on a chain without interacting with it directly.

**Issue**
The issue arises when interacting with an existing network by setting the `SMART_ACCOUNT_ADDRESS`.
The forked network does not update when changes are made to the live networks.
The changes include executing safe transactions through AutoTx.
Because of the above, running AutoTx more than once against a live network does not work because the safe nonce never updates.

**Solution**
Turn the current `start-fork` feature into `start-devnet` (or equivalent) and only use this feature for offline testing where we want to fork a network and execute transactions upon it locally, and persist this between runs. This is useful in development scenarios where you want to run AutoTx offline multiple times, and execute transactions locally without broadcasting to a live network.

In the non-dev scenario, a local fork is not actually needed right now, as we do not currently simulate transactions. In the future when we do, can simulate transactions locally upon the current network (livenet or devnet).

### PR DESCRIPTION
**Pull request by nerfZael**
Closes #162

### CODE REVIEW

File: autotx/utils/ethereum/load_w3.py [hunk 1]
```diff
@@ -1,6 +1,11 @@
...
 from web3 import Web3
-from autotx.utils.ethereum.constants import FORK_RPC_URL
+from autotx.utils.ethereum.constants import DEVNET_RPC_URL
+from autotx.utils.is_dev_env import is_dev_env
 
 def load_w3() -> Web3:
-    web3 = Web3(Web3.HTTPProvider(FORK_RPC_URL))
+    rpc_url = DEVNET_RPC_URL if is_dev_env() else os.getenv("CHAIN_RPC_URL")
+    
+    web3 = Web3(Web3.HTTPProvider(rpc_url))
```
Comment (cbrzn) [inline autotx/utils/ethereum/load_w3.py:9]: we do the same in `get_configuration`, with the difference that we throw is `CHAIN_RPC_URL` is not set - should we add that validation here and use `load_w3` in `get_configuration` ?

File: autotx/utils/ethereum/constants.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 MASTER_COPY_ADDRESS = "0xd9Db270c1B5E3Bd161E8c8503c55cEABeE709552"
-MASTER_COPY_L2_ADDRESS = "0xfb1bffC9d739B8D520DaF37dF666da4C687191EA"
+MASTER_COPY_L2_ADDRESS = "0x3E5c63644E683549055b9Be8653de26E0B4CD36E"
```
Comment (cbrzn) [inline autotx/utils/ethereum/constants.py:2]: what if we make `MASTER_COPY_L2_ADDRESS(ES)` a list and add both values? (the one you added and the one we previous had) - This way we guarantee that we can interact with safe in all the networks we support (the previous value is needed for optimism or sepolia)

**Review by cbrzn — COMMENTED** [review]
these changes makes me think... right now we show the network when we ask something to autotx; if is dev net should we explicitly say that we're working w/the local node? so we avoid confusions? not sure tho...

**Review by dOrgJelli — APPROVED** [review]

File: autotx/utils/ethereum/load_w3.py (hunk 1 above)
Comment (nerfZael) [reply autotx/utils/ethereum/load_w3.py]: I added the check there, but we cant reuse `load_w3` since we also need the `rpc_url` for `EthereumClient`

**Review by cbrzn — APPROVED** [review]

**Pull request merged**